src/Goerli-DssSpell.t.sol

```diff
@@ -2323,131 +2100,196 @@ contract DssSpellTest is DSTest, DSMath {
         jug.drip(_ilk);
         assertEq(clip.kicks(), 0);
         if (_checkLiquidations) {
-            dog.bark(_ilk, address(this), address(this));
+            cat.bite(_ilk, address(this));
             assertEq(clip.kicks(), 1);
         }
 
         // Dump all dai for next run
         vat.move(address(this), address(0x0), vat.dai(address(this)));
     }
 
-    function testVestDAI() public {
+    function checkPsmIlkIntegration(
+        bytes32 _ilk,
+        GemJoinAbstract join,
+        ClipAbstract clip,
+        address pip,
+        PsmAbstract psm,
+        uint256 tin,
+        uint256 tout
+    ) public {
+        DSTokenAbstract token = DSTokenAbstract(join.gem());
+
+        assertTrue(pip != address(0));
+
+        spotter.poke(_ilk);
+
+        // Authorization
+        assertEq(join.wards(pauseProxy), 1);
+        assertEq(join.wards(address(psm)), 1);
+        assertEq(psm.wards(pauseProxy), 1);
+        assertEq(vat.wards(address(join)), 1);
+        assertEq(clip.wards(address(end)), 1);
+
+        // Check toll in/out
+        assertEq(psm.tin(), tin);
+        assertEq(psm.tout(), tout);
+
+        uint256 amount = 1000 * (10 ** token.decimals());
+        giveTokens(token, amount);
+
+        // Approvals
+        token.approve(address(join), amount);
+        dai.approve(address(psm), uint256(-1));
+
+        // Convert all TOKEN to DAI
+        psm.sellGem(address(this), amount);
+        amount -= amount * tin / WAD;
+        assertEq(token.balanceOf(address(this)), 0);
+        assertEq(dai.balanceOf(address(this)), amount);
+
+        // Convert all DAI to TOKEN
+        amount -= amount * tout / WAD;
+        psm.buyGem(address(this), amount);
+        assertEq(dai.balanceOf(address(this)), 0);
+        assertEq(token.balanceOf(address(this)), amount);
+
+        // Dump all dai for next run
+        vat.move(address(this), address(0x0), vat.dai(address(this)));
+    }
+
+    function testCollateralIntegrations() public {
         vote(address(spell));
         scheduleWaitAndCast(address(spell));
         assertTrue(spell.done());
 
-        assertEq(DssVestLike(addr.addr("MCD_VEST_DAI")).cap(), 1 * MILLION * WAD / 30 days);
+        // Insert new collateral tests here
+        checkUNIV2LPIntegration(
```

Comment:
nit: I think we should rename the integration tests for UNIV3

Reply:
done